File: examples/hello.py
"""Minimal hello-world for the dkdc_draw Python bindings.

Requires: `uv run maturin develop` (or `pip install dkdc-draw`).

Run: `uv run python examples/hello.py`
"""

import json
from pathlib import Path

import dkdc_draw

doc_json = dkdc_draw.new_document("hello")
doc = json.loads(doc_json)

doc["elements"] = [
    {
        "type": "Rectangle",
        "id": "box",
        "x": 20.0,
        "y": 20.0,
        "width": 200.0,
        "height": 80.0,
    },
    {
        "type": "Text",
        "id": "label",
        "x": 40.0,
        "y": 50.0,
        "text": "hello, draw",
    },
    {
        "type": "Arrow",
        "id": "arrow",
        "x": 240.0,
        "y": 60.0,
        "points": [{"x": 0.0, "y": 0.0}, {"x": 80.0, "y": 0.0}],
    },
]

doc_json = json.dumps(doc)

Path("hello.svg").write_text(dkdc_draw.export_svg(doc_json))
print("wrote hello.svg")

Path("hello.png").write_bytes(dkdc_draw.export_png(doc_json))
print("wrote hello.png")

out = Path("hello.draw.json")
dkdc_draw.save_document(doc_json, str(out))
print(f"wrote {out}")

File: crates/draw-core/examples/hello.rs
//! A minimal hello-world for draw-core.
//!
//! Builds a small document with a few elements, exports SVG and PNG to
//! the current directory, and prints the paths. Run with:
//!
//!     cargo run --example hello -p dkdc-draw-core

use std::fs;
use std::path::Path;

use draw_core::{
    Document, Element, FreeDrawElement, LineElement, Point, ShapeElement, TextElement, export_png,
    export_svg,
};

fn main() -> anyhow::Result<()> {
    let mut doc = Document::new("hello".into());

    doc.add_element(Element::Rectangle(ShapeElement::new(
        "box".into(),
        20.0,
        20.0,
        200.0,
        80.0,
    )));
    doc.add_element(Element::Text(TextElement::new(
        "label".into(),
        40.0,
        50.0,
        "hello, draw".into(),
    )));
    doc.add_element(Element::Arrow(LineElement::new(
        "arrow".into(),
        240.0,
        60.0,
        vec![Point::new(0.0, 0.0), Point::new(80.0, 0.0)],
    )));
    doc.add_element(Element::FreeDraw(FreeDrawElement::new(
        "squiggle".into(),
        340.0,
        60.0,
        vec![
            Point::new(0.0, 0.0),
            Point::new(10.0, -15.0),
            Point::new(25.0, 10.0),
            Point::new(40.0, -5.0),
            Point::new(55.0, 12.0),
        ],
    )));

    let svg_path = Path::new("hello.svg");
    fs::write(svg_path, export_svg(&doc))?;
    println!("wrote {}", svg_path.display());

    let png_path = Path::new("hello.png");
    fs::write(png_path, export_png(&doc)?)?;
    println!("wrote {}", png_path.display());

    Ok(())
}
